### PR TITLE
Update and consolidate group documentation

### PR DESCRIPTION
**Pull request by frenck**
## Proposed change

Update and consolidate all groups documentation (all old platforms) into a single page.
Added a bit more context on the different types of groups, and updated the documentation for the UI coming to Home Assistant Core 2022.4.

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [x] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: 
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: fixes #

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards][].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

### CODE REVIEW

File: source/_integrations/group.markdown [hunk 1]
```diff
@@ -18,24 +25,184 @@ ha_platforms:
...
-Groups allow the user to combine multiple entities into one.
+The group integration lets you combine multiple entities into a single entity. Entities that are members of a group can be controlled and monitored as a whole.
 
-Check **Developer Tools** -> **States** and browse the **Current entities:** listing for all available entities.
+This can be useful for cases where you want to control, for example, the
+multiple bulbs in a light fixture as a single light in Home Assistant.
+
+Home Assistant can group multiple binary sensors, covers, fans, light, media players as a single entity, with the option of hiding the individual member entities.
+
+{% include integrations/config_flow.md %}
```
Comment (balloob) [inline source/_integrations/group.markdown:35]: Can we make a `helpers/config_flow.md` ? I want us to add a new helper My Button (but we didn't get to it yet). We can do this in another PR too.

File: source/_integrations/group.markdown [hunk 2]
```diff
@@ -18,24 +25,184 @@ ha_platforms:
...
+
+## Group behavior
+
+By default, when any group member entity is `on`, the group will also be `on`. A complete overview of how groups behave:
+
+- The group state is `unavailable` if all group members are `unavailable`.
+- Otherwise, group state is `unknown` if all group members are `unknown`.
+- Otherwise, group state is `on` if at least one group member is `on`.
```
Comment (balloob) [inline source/_integrations/group.markdown]: Can we move this to be the first entry in the list

**Review by balloob — DISMISSED** [review]

File: source/_integrations/group.markdown (hunk 1 above)
Comment (frenck) [reply source/_integrations/group.markdown:35]: Let's do that for all the helpers when its set and ready to go :)

File: source/_integrations/group.markdown (hunk 2 above)
Comment (mdegat01) [reply source/_integrations/group.markdown]: Probably for the section below too. Seems like it should first say how `on` and `off` works, then talk about the more exceptional cases of `unknown` and `unavailable`

File: source/_integrations/group.markdown [hunk 3]
```diff
@@ -18,24 +25,184 @@ ha_platforms:
...
+## Group behavior
+
+By default, when any group member entity is `on`, the group will also be `on`. A complete overview of how groups behave:
+
+- The group state is `on` if at least one group member is `on`.
+- Otherwise, the group state is `unavailable` if all group members are `unavailable`.
+- Otherwise, the group state is `unknown` if all group members are `unknown`.
+- Otherwise, the group state is `off`.
+
+Some groups, like the binary sensors and lights, allow you set "All entities" option. When enabled, this behavior is inverted, and all members of the group have to be `on` for the group to turn `on` as well. A complete over of how groups behave when the "All entities" option is enabled:
```
Comment (emontnemery) [inline source/_integrations/group.markdown]:
```suggestion
Some groups, like the binary sensors and lights, allow you set the "All entities" option. When enabled, this behavior is inverted, and all members of the group have to be `on` for the group to turn `on` as well. A complete overview of how groups behave when the "All entities" option is enabled:
```

File: source/_integrations/group.markdown [hunk 4]
```diff
@@ -18,24 +25,184 @@ ha_platforms:
...
+
+- The group state is `off` if at least one group member is `off`.
+- Otherwise, the group state is `unavailable` if all group members are `unavailable`.
+- Otherwise, the group state is `unknown` if at least one group member is `unknown` or `unavailable`.
+- Otherwise, the group state is `on`.
+
+## YAML Configuration
+
+Alternatlively, this integration can be configured and set up manually via YAML
+instead. Here are example of how to groups when using the `configuration.yaml` file.
```
Comment (emontnemery) [inline source/_integrations/group.markdown]:
```suggestion
instead. Here are example of how to configure groups when using the `configuration.yaml` file.
```

**Review by emontnemery — APPROVED** [review]
